### ISSUE TITLE
Consecutive BatchNorm1d is not equivalent to the last one

### ISSUE THREAD

**Issue by thomasw21** [pytorch/pytorch#83747]
### 🐛 Describe the bug

AFAIK, consecutive BatchNorm1d should be equivalent to the last BatchNorm1d, but the output doesn't seem to be the case. Am I missing something?

```python
from torch.nn import BatchNorm1d
from torch import randn

x = randn(2,3)

l1 = BatchNorm1d(3, momentum=None, track_running_stats=None)
l1.weight.data = randn(3)
l1.bias.data = randn(3)
l2 = BatchNorm1d(3, momentum=None, track_running_stats=None)
l2.weight.data = randn(3)
l2.bias.data = randn(3)

print("///// compare l2(l1(x)) and l2(x) ////")
print(l2(l1(x)))
print(l2(x)) # not the same, but almost the same if you permute the values around

print("///// compare l1(l2(x)) and l1(x) ////")
print(l1(x))
print(l1(l2(x))) # not the same, but almost the same if you permute the values around
```

### Versions

PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.5 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.21.3
Libc version: N/A

Python version: 3.8.12 | packaged by conda-forge | (default, Sep 29 2021, 19:21:23)  [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.5-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.812
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.1
[pip3] torch==1.12.0
[conda] numpy                     1.23.1                   pypi_0    pypi
[conda] torch                     1.12.0                   pypi_0    pypi

**Comment by ptrblck** [comment]
In your code snippet you are initializing the trainable, affine parameters with different `randn` calls in both batchnorm layers, such that the output would be calculated as:
```python
out1 = (x - mean_x) / std_x * weight1 + bias1
out2 = (out1 - mean_out1) / std_out1 * weight2 + bias2
```
and I would thus expect to see different results.

**Comment by thomasw21** [comment]
I think the argument I was trying to make is that 
```
(x - mean_x) / std_x == (out1 - mean_out1) / std_out1 # Should be True
```
But actually it's only true when `weight` are positive.

Thanks!
